its 4:10am help me

Rick10101221 opened the pull request:
math midterm tmmr havent studied :)

Changes:
- All routes are filled-in (with the exception of corkboard and hugs, whose models need to be completed and tested)
- Added TODO tags on the top of routes (in users, notifications, and hugs) that are broken and VERIFIED tags for ones that are fully completed and operational
- Added a fully-fledged suggestion of code in buildCorkboard that will synchronously build the corkboard (with the exception of the date_time field, which cannot be converted from TimeStamp Firestore object to JSON, so it is not being passed back at all. This needs to be looked into)

Rick10101221 committed:
changes

Rick10101221 committed:
merge conflict

Rick10101221 committed:
added TODO tags for a few routes and fixed a few models

Rick10101221 committed:
added a few more routes

Rick10101221 committed:
added complete routes with all current models

terryzfeng commented on backend/model/Notifications.js:
```diff
@@ -50,40 +30,40 @@ const NotificationsAPI = {
+        const newHug = {
+          type : "hug",
+          hug_id : hug_id,
+          date_time : dateTime,
```
This needs to be changed to a firebase timestamp object

terryzfeng commented on backend/routes/notificationsRoutes.js:
```diff
@@ -53,51 +54,85 @@ async function notificationCollectionLength(req, uid) {
+        friendPfp: userResponse.profile_pic
+      }
+      notifResponse[i]['friendInfo'] = newUser;
+    }
```
We should be doing this work in the body of the function, not the route

terryzfeng commented on backend/model/Corkboard.js:
```diff
@@ -8,47 +8,82 @@ require("firebase/auth");
+    buildCorkboard: async function(uid) {
+        // const currUser = firebase.auth().currentUser;
+
         let pinnedHugsList = [];
```
@EmanSherif

terryzfeng commented on backend/model/Notifications.js:
```diff
@@ -11,37 +11,17 @@ const users = db.collection("users");
+        const notificationSnapshot = await notificationCollection.get();
+        let notifications = [];
+        notificationSnapshot.forEach(doc => {
+          notifications = [...notifications, doc.data()];
```
Notifications should be handling the logic, so the route is just calling this and passing back the return

evanjserrano commented on backend/model/Users.js:
```diff
@@ -84,6 +84,16 @@ const UsersAPI = {
     return userProfile;
   },
 
+  checkUserExists: async function(email) {
+    let success = false;
+    const userRef = usersCollection.where('email', '==', email);
+    const user = await userRef.get();
+    if (!user.exists) {
+      success = true;
+    }
+    return { out: status };
+  },
+
```
Good function. Need to update documentation

terryzfeng replied:
Do we have an email field?

evanjserrano replied:
Oh oops we're not storing the email so this should be with something else.. maybe use authentication? I can work on this if you need

evanjserrano replied:
I take it back. Bad function. ;) jk

evanjserrano reviewed (dismissed):
Fix or change that Users.js func

Rick10101221 committed:
in response to pr comments

Rick10101221 commented:
Changes
- Scratched checkUserExists(email)
- Notifications now have firestore timestamp objects instead of JS Date objects
- Moved notificationRoutes.js model logic from routes to Notifications.js
- Added route skeletons to friendsRoutes.js in prep for Friends.js model merge
- Added NOT TESTED and BROKEN tags at the top of routes for Friends, Corkboards, Accounts

terryzfeng reviewed (approved):
Just realized this was from the backend/hugs branch. All good from me.

tyusliu reviewed (approved):
I see what you did in Hugs, lgtm. Only thing is in hugsRoutes the comment says we are still trying to figure out blobs? or we sticking with base 64 something?

terryzfeng committed:
Updated code and Firestore to match DB Schema

Rick10101221 committed:
finished frontend API. my backend work is done for today and tmmr pog

Rick10101221 committed:
Merge branch 'backend/hugs' of https://github.com/evansuong/fouro-application into backend/hugs

vukRadovanovic commented on backend/model/Users.js:
```diff
@@ -139,6 +139,7 @@ const UsersAPI = {
   },
 
   uploadUserProfilePicture: async function (uid, file) {
+    console.log('here');
```
I believe that whatever message is logged should be a little bit more descriptive.

vukRadovanovic reviewed (commented):
Looks good!  Just that minor detail of what's being console.logged in Users.js that can be easily resolved.
